esp-hal/src/uart.rs

```diff
@@ -1899,10 +1899,21 @@ mod asynch {
         }
     }
 
-    impl<T> UartTx<'_, T, Async>
+    impl<'d, T> UartTx<'d, T, Async>
     where
-        T: Instance,
+        T: Instance + 'd,
     {
+        /// Create a new UART TX instance in [`Async`] mode.
+        pub fn new_async<TX: OutputPin>(
```

Comment:
These also need to bind the relevant interrupt to the async handler too. The interrupt is shared between tx and rx, but it should be fine to set it on both as it will be set to the same interrupt.